### PR TITLE
Using RxUI and WPF for AddPackage form #33

### PR DESCRIPTION
**Pull request by TheAngryByrd**
This is the initial pull request for #33.

Still nothing pretty to look at but it's a start.

addpackageform

addpackagebutton

**Warning**: my VS seems to be shoving tons of things in the csproj.  Is there anything I can do to fix that?

### CODE REVIEW

**Comment by forki** [comment]
Wow that was fast. Very cool. 

Regarding csproj file: judging by the changes in paket.lock I think it's expected to have such a big footprint in the project. Seems we added a huge dependency.

File: paket.dependencies [hunk 1]
```diff
@@ -4,9 +4,10 @@ nuget FAKE
 nuget Paket.Core
 nuget FSharp.Formatting
 nuget VSSDK.GraphModel.12 copy_local: false
-nuget VSSDK.Text.12 copy_local: false
```
Comment (forki) [inline paket.dependencies:7]: Why did you remove that one?

File: src/Paket.VisualStudio/Commands/AddPackageProcess.cs [hunk 2]
```diff
@@ -1,47 +1,48 @@
...
 namespace Paket.VisualStudio.Commands
 {
     public class AddPackageProcess
     {
+        public static Task<string[]> SearchPackagesByName(string name)
+        {
+           return FSharpAsync.StartAsTask(NuGetV3.FindPackages(FSharpOption<Paket.Utils.Auth>.None, Constants.DefaultNugetStream,
+                    name, 1000),
+                FSharpOption<TaskCreationOptions>.None,
+                FSharpOption<CancellationToken>.None);
```
Comment (forki) [inline src/Paket.VisualStudio/Commands/AddPackageProcess.cs:23]: I think we should pass the cancellation token of the new task

File: paket.dependencies (hunk 1 above)
Comment (TheAngryByrd) [inline paket.dependencies:7]: Looks like 'paket simplify' might have hit that one.  I'll add it back in when I get home.

File: src/Paket.VisualStudio/Commands/AddPackageProcess.cs (hunk 2 above)
Comment (TheAngryByrd) [inline src/Paket.VisualStudio/Commands/AddPackageProcess.cs:23]: Looks like I was refactoring and didn't finish! :scream:  This is what I get for doing PR this early :)

**Comment by forki** [comment]
very cool. will test it out

File: src/Paket.VisualStudio/Commands/AddPackageProcess.cs (hunk 2 above)
Comment (forki) [inline src/Paket.VisualStudio/Commands/AddPackageProcess.cs:23]: no worries. I merged it anyways since it's already much better than before. Please send new PR if you want to improve it ;-)

**Comment by forki** [comment]
I think it should be possible to add a package even if I didn't select one.
There are cases when we can't find the package in autocomplete, but the package is in a private feed. Then it should be possible to just write the name and add it anyways.

**Comment by hmemcpy** [comment]
Hmm, in this case I'd prompt for this with a message box. Ask the user if
they are sure. When in Rome... (Windows :))

File: src/Paket.VisualStudio/Commands/AddPackageProcess.cs [hunk 3]
```diff
@@ -1,47 +1,48 @@
...
+
+            Action<NugetResult> addPackageToDependencies = result =>
             {
-                if (listBox.SelectedItem == null)
-                    return;
-                var packageName = listBox.SelectedItem.ToString();
-                frm.Close();
+                var packageName = result.PackageName;
+                secondWindow.Close();
                 Application.DoEvents();
```
Comment (forki) [inline src/Paket.VisualStudio/Commands/AddPackageProcess.cs:46]: wo we still need this?

**Comment by forki** [comment]
yep a message box could be ok, but at leasrt we should allow it.
I mean if the package doesn't exist the add process will fail anyways.
